### PR TITLE
remove unset keys in place instead of throwing away the whole dict and rebuilding

### PR DESCRIPTION
**Pull request by pjz**

### CODE REVIEW

File: etcd3/apis/kv.py [hunk 1]
```diff
@@ -197,7 +197,9 @@ def range(
             "min_create_revision": min_create_revision,
             "max_create_revision": max_create_revision
         }
-        data = {k: v for k, v in data.items() if v is not None}
+        for k, v in list(data.items()):
```
Comment (Revolution1) [inline etcd3/apis/kv.py:200]: here a new list created, I'm not sure if it's better than just create a new dict.

**Comment by pjz** [comment]
ugh, now I'm not sure if the list-creation overhead is any better than the dict-creation overhead. 
Maybe ignore this one until it gets some microbenchmarks on it.

File: etcd3/apis/kv.py (hunk 1 above)
Comment (pjz) [reply etcd3/apis/kv.py:200]: hah, I think I was typing out this exact thing when you were :)

**Comment by pjz** [comment]
Okay, no faster way to do this, looks like. ignore this.

**Pull request closed without merging**